frontend/src/utils/auth-utils.js

```diff
@@ -0,0 +1,71 @@
+import config from "../../config/config";
+
+export class AuthUtils {
+    static accessTokenKey = 'accessToken';
+    static refreshTokenKey = 'refreshToken';
+    static userKey = 'user';
+
+
+    static async processUnauthorizedResponse() {
+        const refreshToken = localStorage.getItem(this.refreshTokenKey);
+        if (refreshToken) {
+            const response = await fetch(config.api + '/refresh', {
+                method: 'POST',
+                headers: {
+                    'Content-type': 'application/json',
+                    'Accept': 'application/json',
+                },
+                body: JSON.stringify({refreshToken: refreshToken})
+            });
+
+            if (response && response.status === 200) {
+                const result = await response.json();
+                if (result && !result.error) {
+                    this.setTokens(result.accessToken, result.refreshToken);
+                    return true;
+                }
+            }
+        }
+        this.removeAuthInfo();
+        location.href = '/login';
+
+        return false;
+    }
+
+    static setTokens(accessToken, refreshToken) {
+        localStorage.setItem(this.accessTokenKey, JSON.stringify((accessToken)));
+        localStorage.setItem(this.refreshTokenKey, JSON.stringify((refreshToken)));
+    }
+
+    static setUser(user) {
+        localStorage.setItem(this.userKey, JSON.stringify((user)));
+    }
+
+    static getUserInfo(){
+        const userInfo = localStorage.getItem('userKey');
```

Comment:
`
const userInfo = localStorage.getItem(this.userKey)
```suggestion
        const userInfo = localStorage.getItem(this.userKey);
```
`